server/main.js

```diff
@@ -123,40 +125,43 @@ Meteor.publish('tabular_getInfo', function (tableName, selector, sort, skip, lim
     const paths = getSearchPaths(table);
     const newSort = transformSortArray(findOptions.sort);
 
-    filteredRecordIds = table.searchCustom(
+    filteredRecordIdsPromise = Promise.resolve( table.searchCustom(
       this.userId,
       newSelector,
       tokens,
       paths,
       newSort,
       skip,
       limit
-    );
+    ));
   } else {
     filteredCursor = table.collection.find(newSelector, findOptions);
-
-    filteredRecordIds = filteredCursor.map((doc) => doc._id);
+    filteredRecordIdsPromise = filteredCursor.mapAsync((doc) => doc._id);
     countCursor = table.collection.find(newSelector, { fields: { _id: 1 } });
   }
   let fakeCount;
   //if the number of results is greater than the limit then we need to remove the last one
   //and set the fake count to the limit + skip
-  //limit can be null - so don't process it if it is
-  if (limit && filteredRecordIds.length > limit) {
-    //keep only first $limit records in filteredRecordIds
-    fakeCount = filteredRecordIds.length + skip;
-    filteredRecordIds.splice(limit, filteredRecordIds.length - limit);
-  } else {
-    fakeCount = filteredRecordIds.length + skip;
-  }
+  filteredRecordIdsPromise.then(( res ) => {
+    //console.debug( 'filteredRecordIds resolved with res=', res );
+    filteredRecordIds = res;
+    //limit can be null - so don't process it if it is
+    if (limit && filteredRecordIds.length > limit) {
+      //keep only first $limit records in filteredRecordIds
+      fakeCount = filteredRecordIds.length + skip;
+      filteredRecordIds.splice(limit, filteredRecordIds.length - limit);
+    } else {
+      fakeCount = filteredRecordIds.length + skip;
+    }
+  });
   let recordReady = false;
-  let updateRecords = () => {
+  let updateRecords = async () => {
     let currentCount;
     if (!table.skipCount) {
       if (typeof table.alternativeCount === 'function') {
         currentCount = table.alternativeCount(newSelector);
```

Comment:
I'm worried what people might typically do inside here. We don't use this but if people are accessing collections we'll need to make it async friendly.

Reply:
Sorry I should have been more specific here. I meant the alternativeCount function and what they could possibly call in that, so it would need `await table.alternativeCount` ?